### PR TITLE
Modify the document about operator==

### PR DESCRIPTION
**Pull request by dota17**

### CODE REVIEW

**Comment by dota17** [comment]
Related issues #1915  #1953

File: README.md [hunk 1]
```diff
@@ -1011,6 +1012,34 @@ std::vector<std::uint8_t> v_ubjson = json::to_ubjson(j);
...
+This JSON library is very flexible in usage, you can customize and implement you desired functions in your own projects, and can also self-defined data type that you want to parse, or handle some edge cases by yourselves.
+
+#### Custom comparison function
+
+As we know, the result of comparing two floating point numbers may be uncertain, which
+depends on the system architecture, the compiler, and the absolute/relative error value we can acccpt.
+For this reason, you can self-defined function according to specific scenarios, like
+
+```cpp
+bool is_equal(const_reference lhs, const_reference rhs)
```
Comment (nlohmann) [inline README.md]: I am not sure whether the added section to the README actually adds value, as the [documentation of `operator==`](https://nlohmann.github.io/json/classnlohmann_1_1basic__json_a122640e7e2db1814fc7bbb3c122ec76e.html#a122640e7e2db1814fc7bbb3c122ec76e) already describes the current behavior and provides and example for a different comparison.

That said, I think the code snippet is better suited in the documentation of `operator==` while letting the README unchanged.

File: include/nlohmann/json.hpp [hunk 2]
```diff
@@ -5637,13 +5637,14 @@ class basic_json
       their stored values are the same according to their respective
       `operator==`.
     - Integer and floating-point numbers are automatically converted before
-      comparison. Note than two NaN values are always treated as unequal.
+      comparison.
+    - Two NaN values are always treated as unequal.
```
Comment (nlohmann) [inline include/nlohmann/json.hpp]: I would like to keep this in one item, as NaN comparison is related to the previous sentence.

File: README.md (hunk 1 above)
Comment (dota17) [reply README.md]: > That said, I think the code snippet is better suited in the documentation of operator== while letting the README unchanged.

Fine. I will move it.

File: README.md [hunk 3]
```diff
@@ -34,6 +34,7 @@
   - [Conversions to/from arbitrary types](#arbitrary-types-conversions)
   - [Specializing enum conversion](#specializing-enum-conversion)
   - [Binary formats (BSON, CBOR, MessagePack, and UBJSON)](#binary-formats-bson-cbor-messagepack-and-ubjson)
+  - [User self-defined cases](#User-self-defined-cases)
```
Comment (nlohmann) [inline README.md]: This line should be removed, too.

File: README.md [hunk 4]
```diff
@@ -1011,7 +1012,6 @@ std::vector<std::uint8_t> v_ubjson = json::to_ubjson(j);
 json j_from_ubjson = json::from_ubjson(v_ubjson);
 ```
 
-
```
Comment (nlohmann) [inline README.md]: Please undo all changes in README.md.

File: README.md (hunk 3 above)
Comment (dota17) [reply README.md]: Ok, I'll do it today.

**Comment by nlohmann** [comment]
Closes #1915.

**Comment by nlohmann** [comment]
Thanks!

**Comment by nlohmann** [comment]
* * *

### 🔖 Release item

This issue/PR will be part of the next release of the library. This template helps preparing the release notes.

#### Type

- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] ⚡️ Improvement
- [x] 🔨 Further Change
- [ ] 🔥 Deprecated function

#### Description

- Extended documentation of `operator==`. #1915 #1984 

* * *

**Pull request merged**